main.cpp

```diff
@@ -188,8 +196,17 @@ struct IntType
     IntType& multiply(int myNumber);
     IntType& divide(int myNumber);
 
+    IntType& pow(int myNumber);
+    IntType& pow(const IntType& myNumber);
+    IntType& pow(const FloatType& myNumber);
+    IntType& pow(const DoubleType& myNumber);
+
 private:
     int* value = nullptr;
+    IntType& powInternal(int myNumber);
+    IntType& powInternal(const IntType& myNumber);
+    IntType& powInternal(const FloatType& myNumber);
+    IntType& powInternal(const DoubleType& myNumber);
```

Comment:
There should only be one powInternal function

- [ ] remove these extra `powInternal` functions